tests/conftest.py

```diff
@@ -2,5 +2,27 @@
 
 import pytest
 
+
 # Ensure support module assertions provide failure details
 pytest.register_assert_rewrite("tests.support")
+
+# Session-scoped fixture for required model loading
+import asyncio
+import sys
+
+@pytest.fixture(scope="session", autouse=True)
+def load_required_models():
+    """Load required models at the start of the test session."""
+    # Only run if LM Studio is accessible
+    try:
+        from tests.load_models import reload_models
+        asyncio.run(reload_models())
+    except Exception as e:
+        print(f"[Fixture] Skipping model loading: {e}", file=sys.stderr)
```

Comment:
Some environments consider printing anything to stderr to be a test failure in its own right, so ideally we wouldn't even try to load the models when `not lmstudio` is part of the test marker filtering (or all the other model using test cases are otherwise filtered out).

This suggests that rather than `autouse=True`, we want something along the lines of the "marker to fixture usage" approach suggested in the [this `pytest` issue comment](https://github.com/pytest-dev/pytest/issues/6115):

```python
def pytest_itemcollected(item):
    if item.get_closest_marker("lmstudio") is not None:
        item.applymarker(pytest.mark.usefixtures("load_required_models"))
```